src/jdk.compiler/share/classes/com/sun/tools/javac/code/Symbol.java

```diff
@@ -1516,11 +1513,18 @@ public RecordComponent createRecordComponent(JCVariableDecl var, List<JCAnnotati
                     toRemove = rc;
                 }
             }
+            return toRemove;
+        }
+
+        /* creates a record component if non is related to the given variable and recreates a brand new one
+         * in other case
+         */
+        public RecordComponent createRecordComponent(RecordComponent existing, JCVariableDecl var, List<JCAnnotation> annotations) {
             RecordComponent rc = null;
-            if (toRemove != null) {
+            if (existing != null) {
                 // Found a record component with an erroneous type: remove it and create a new one
```

Comment:
I don't think this comment is valid anymore as we are removing an existing record component regardless